test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -169,4 +177,19 @@ private boolean verifyEventEquality(Object expectedEvent, Object actualEvent) {
         }
         return true;
     }
+
+    private boolean verifyMetaDataEquality(MetaData expectedMetaData, MetaData actualMetaData) {
```

Comment:
I agree with @smcvb that it doesn't make much sense to check for equality of the meta data type here. In fact, Meta Data is just a Map implementation and it doesn't matter which implementation is really used. 
When comparing meta-data, instead of checking field equality of the entire MetaData object, I would first validate if all the expected meta data keys are available in the actual meta data. Then, for each key, check if the values are equal. Given the meta data values should be value objects, it is fine to just compare then using Objects.equals().